daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -465,12 +488,12 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
   /**
    * Creates a new instance of a variable with default value
    */
-  def newVariableInstance(vrd: VariableRuntimeData, defaultValue: DataValuePrimitive) = {
+  def newVariableInstance(vrd: VariableRuntimeData, defaultValue: DataValuePrimitiveNullable) = {
     val varQName = vrd.globalQName
     val variableInstances = vTable.get(varQName)
     Assert.invariant(variableInstances.isDefined)
 
-    variableInstances.get += vrd.createVariableInstance(VariableUtils.convert(defaultValue.getAnyRef.toString, vrd, vrd))
+    variableInstances.get += vrd.createVariableInstance(defaultValue)
   }
```

Comment:
I would actually prefer we drop this function all together. If I want to find all places where we set a defaultValue, I want to be able to grep for ``setDefaultValue``. Having to also grep for newVarialbeInstance where we pass in an extra paramter is more difficult. Is there any reason why uses of this can't be replaced with
```scala
val v = newVariableInstance(...)
v.setDefaultValue(...)
```

Reply:
Makes sense to me, will change.